nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-headless-server/src/main/java/org/apache/nifi/headless/HeadlessNiFiServer.java

```diff
@@ -137,7 +141,8 @@ public void preDestruction() throws AuthorizerDestructionException {
                     bulletinRepository,
                     variableRegistry,
                     flowRegistryClient,
-                    extensionManager);
+                    extensionManager,
+                    statusHistoryRepository);
```

Comment:
This is not something we want to introduce with this change. By this change status history would be involved into headless NiFi (also with hard coded implementation) which might come with unforeseen consequences. Please find a solution which does not include changing headless.

Reply:
I can think of two solutions now, but we can talk about it if needed. And any better ideas are appreciated.
1. A default interface method would do it.
2. I can try to move the StatusHistoryDumpFactory::getStatusHistoryDumpFactory to a separate interface which only the JettyServer would implement. In the BootstrapListener with an instanceOf check and casting, the method would be accessible.

Reply:
I double checked and I have failed to recognize that headless used a factory method which provides a `FlowController` instance with `StatusHistoryRepository` implicitely. The explicit declaration misleaded me, however hardcoding the `VolatileComponentStatusRepository` is still an issue. It should use the instance created by the Spring context.

Reply:
I believe we should first determine if we require this functionality on the https://nifi.apache.org/minifi/ server as well. The StandardMiNiFiServer's ancestor is the HeadlessNiFiServer. What are your thoughts on just returning null or throwing an UnsupportedOperationException?
I realize that constructing a VolatileComponentStatusRepository will not do, but this capability might not be required here. The DecomissionTask, for example, was added to the NiFi server interface earlier this year, and the headless implementation simply returns null. The spring application context is not utilized at all in the class, and no injection is used for object creation, which might be for a cause.

Reply:
I would believe we should still not having a hardcoded implementation and maybe the best way would be to depend on the `nifi.properties` and mimic the way `StatusHistoryRepositoryFactoryBean` works. What I am thinking about is to extract the relevant parts of the bean creation (so the part which is not depending on Spring) and use that logic here as well. I need to double check this idea but please check it too if this would solve the issue.